src/globus_cli/parsing/param_types/comma_delimited.py

```diff
@@ -18,7 +33,18 @@ def convert(self, value, param, ctx):
         # it means that if you take
         # `--foo={",".join(original)}`, you will get a value equal to
         # `original` back if `original=[]` (but not if `original=[""]`)
-        elif value == "":
-            return []
-        else:
-            return value.split(",")
+        resolved = value.split(",") if value else []
+
+        if self.convert_values is not None:
+            resolved = [self.convert_values(x) for x in resolved]
+
+        if self.choices is not None:
+            bad_values = [x for x in resolved if x not in self.choices]
+            if bad_values:
+                self.fail(
+                    f"the values {bad_values} were not valid choices",
```

Comment:
Might be nice to have the choices in the error output?
But also could make the error wayy too long, so nbd